tasks/__init__.py

```diff
@@ -47,8 +47,8 @@ def _init_event_producer(config):
         return NullProducer()
 
 
-def emit_event(e):
-    producer.send(cfg.event_topic, value=e.encode("utf-8"))
+def emit_event(e, id):
+    producer.send(cfg.event_topic, value=e.encode("utf-8"), key=str(id).encode("utf-8"))
```

Comment:
Assuming that this is a generic method that expects a message key string, I’d just treat the value as a string. It’s the caller’s responsibility to pass a string.

```suggestion
    k = key.encode("utf-8") if key else None
    v = e.encode("utf-8")
    producer.send(cfg.event_topic, key=k, value=v)
```